### DISCUSSION TITLE
Temperature motor BBS02B

### DISCUSSION THREAD

**Discussion by kb79000** [danielnilsson9/bbs-fw#134] [Q&A]
Hello with the heat that arrives I very regularly check the engine temperature because impossible to control that of the controller on BBS02B.
I try that my engine does not exceed 43 degrees.
I did not understand well there is a clamping of the engine if a temperature is exceeded?
Thank you very much for your feedback 😉

**Comment by danielnilsson9** [comment]
No motor core temperature sensor on the BBS02. Only a temperature sensor in the controller on the MOSFETS.
Temperature limit currently set to 85C here https://github.com/danielnilsson9/bbs-fw/blob/cc716b2682b34b5024f14beda708d73e61db06dd/src/firmware/fwconfig.h#L38

Limit applies to both temperature sensors on the BBSHD.

**Reply by kb79000** [reply]
I think that's huge for degrees celsius, right?  because I regularly check my temperature and make sure that it does not exceed 45° because the outside protection is already very hot when the temperature is raised

**Reply by dav0000000** [reply]
I've seen my BBSHD controller hit the low 80's and start limiting many times without any issues... yet.

**Reply by danielnilsson9** [reply]
I think that is a pretty safe limit. Both windings and MOSFETS below 100C should be no issue.

**Reply by kb79000** [reply]
I think for the BBSHD the temperature is not necessarily an issue, but for my BBS02B I think the controller is more sensitive to heat Are the temperature limits the same on all three models?  THANKS

**Reply by danielnilsson9** [reply]
Yes, 85 for everything. 100C should be fine on BBS02 controller too or why not? IRFB3077 mosfets in BBS02 can handle 175C.  Problem is windings on BBS02, since there is no monitoring they could potentially get too hot and get damaged without controller reaching limiting temperature of 85. Nothing we can do about that without modifying motor/controller.

You will need to compile your own version if you want to change the limit. https://github.com/danielnilsson9/bbs-fw/wiki/Build-from-Source

**Reply by kb79000** [reply]
Ok, I've heard quite a few people complaining about fried their controller.  I just wanted to make sure that all possibilities to avoid this incident had been taken into account 😉

**Reply by danielnilsson9** [reply]
Ok, I cannot leave any guarantees, everything can happen. But I'm pretty sure this firmware limits temperature long before the original firmware did (if ever?). Temperature was initially set lower (65) but many people were complaining and I figured 85 should still be a pretty safe limit (if sensor is working correctly).
